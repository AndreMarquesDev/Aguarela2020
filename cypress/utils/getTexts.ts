import { TextsInterface, textsPt, textsEn } from '../../utils/texts';
import { Locale } from '../../utils/locales';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === 'pt') {
        return textsPt;
    }

    return textsEn;
};
