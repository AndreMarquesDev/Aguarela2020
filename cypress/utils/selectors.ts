const baseUrl = 'http://localhost:3000/';
const baseUrlPt = `${baseUrl}pt/`;
const baseUrlEn = `${baseUrl}en/`;

export const urls = {
    // eslint-disable-next-line id-length
    pt: {
        homepage: `${baseUrlPt}homepage`,
        about: `${baseUrlPt}about`,
        projects: `${baseUrlPt}projects`,
        services: `${baseUrlPt}services`,
        contact: `${baseUrlPt}contact`,
    },
    // eslint-disable-next-line id-length
    en: {
        homepage: `${baseUrlEn}homepage`,
        about: `${baseUrlEn}about`,
        projects: `${baseUrlEn}projects`,
        services: `${baseUrlEn}services`,
        contact: `${baseUrlEn}contact`,
    },
};
