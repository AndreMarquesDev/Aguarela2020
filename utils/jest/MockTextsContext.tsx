import React, { ReactNode } from 'react';
import { TextsContext } from '../../components/context/TextsContext';
import { textsEn, textsPt } from '../texts';

interface MockTextsContextProps {
    isEnglish: boolean;
    children: ReactNode;
}

export const MockTextsContext = ({ isEnglish, children }: MockTextsContextProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: isEnglish ? textsEn : textsPt,
            }}
        >
            {children}
        </TextsContext.Provider>
    );
};
