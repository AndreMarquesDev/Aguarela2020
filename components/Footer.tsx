/* eslint-disable jsx-a11y/anchor-is-valid */
import React, { FC, useContext } from 'react';
import Link from 'next/link';
import TextsContext from './context/TextsContext';
import { Page } from '../utils/pages';

interface FooterProps {
    language: string;
}

const Footer: FC<FooterProps> = ({ language }) => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <div className="wrapper">
                <p>
                    {texts.footerInfo}
                    {' '}
                    <a href="https://www.andremarquesdev.com" rel="noreferrer" target="_blank">Andr√© Marques</a>
                </p>
                <ul>
                    <li><a href="https://www.instagram.com/aguareladigital" rel="noreferrer" target="_blank">I</a></li>
                    <li><a href="https://www.facebook.com/aguareladigitalagency" rel="noreferrer" target="_blank">F</a></li>
                    <li>
                        <Link href={`/${language}/${'contact' as Page}`} prefetch={false}>
                            <a>
                                M
                            </a>
                        </Link>

                    </li>
                </ul>
            </div>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    div {
                        display: flex;
                        justify-content: space-between;
                        align-items: center;
                        background: $grey;
                        padding-top: 12.5rem;
                        padding-bottom: 12.5rem;
                    }

                    p {
                        @include fontXS(uppercase);
                    }

                    ul {
                        display: flex;
                        justify-content: space-between;
                        align-items: center;

                        li {
                            width: 35rem;
                            height: 35rem;
                            position: relative;
                            text-align: center;
                            margin: 0 25rem;
                            color: $white;

                            &:before {
                                content: '';
                                width: 100%;
                                height: 100%;
                                position: absolute;
                                top: 0;
                                left: 0;
                                background: $black;
                                border-radius: 50%;
                            }

                            a {
                                display: block;
                                position: relative;
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};

export default Footer;
