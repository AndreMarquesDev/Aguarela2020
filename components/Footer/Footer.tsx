import React, { useContext } from 'react';
import Link from 'next/link';
import { FaInstagram, FaFacebookF, FaEnvelope } from 'react-icons/fa';
import { TextsContext } from '../context/TextsContext';
import { Page } from '../../utils/pages';
import { Locale } from '../../utils/locales';
import { footerDataTestId } from '../../utils/dataTestIds';
import {
    aguarelaDigitalFacebookUrl,
    aguarelaDigitalInstagramUrl,
    andreMarquesDevWebsiteUrl,
} from '../../utils/urls';

export interface FooterProps {
    language: Locale;
}

export const Footer = ({ language }: FooterProps): JSX.Element => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <footer data-testid={footerDataTestId}>
                <div className="wrapper">
                    <p>
                        {texts.footerInfo}{' '}
                        <a
                            className="animatedLink animatedLinkWhite"
                            href={andreMarquesDevWebsiteUrl}
                            rel="noreferrer"
                            target="_blank"
                        >
                            Andr√© Marques
                        </a>
                    </p>

                    <ul>
                        <li>
                            <a
                                aria-label="Aguarela instagram"
                                href={aguarelaDigitalInstagramUrl}
                                rel="noreferrer"
                                target="_blank"
                            >
                                <FaInstagram />
                            </a>
                        </li>
                        <li>
                            <a
                                aria-label="Aguarela facebook"
                                href={aguarelaDigitalFacebookUrl}
                                rel="noreferrer"
                                target="_blank"
                            >
                                <FaFacebookF />
                            </a>
                        </li>
                        <li>
                            <Link href={`/${language}/${'contact' as Page}`} prefetch={false}>
                                {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
                                <a aria-label="Go to contact page">
                                    <FaEnvelope />
                                </a>
                            </Link>
                        </li>
                    </ul>
                </div>
            </footer>

            <style jsx>
                {`
                    @import './styles/_vars.scss';

                    footer {
                        background: $blue;
                        padding-top: 12.5rem;
                        padding-bottom: 12.5rem;
                    }

                    .wrapper {
                        display: flex;
                        justify-content: space-between;
                        align-items: center;

                        @include tablet {
                            flex-direction: column-reverse;
                        }
                    }

                    p {
                        @include fontXS($white, uppercase);

                        @include tablet {
                            margin-top: 25rem;
                            text-align: center;
                        }
                    }

                    ul {
                        display: flex;
                        justify-content: space-between;
                        align-items: center;

                        li {
                            width: 35rem;
                            height: 35rem;
                            position: relative;
                            text-align: center;
                            margin: 0 calc(25rem / 2);
                            color: $blue;

                            &:before {
                                content: '';
                                width: 100%;
                                height: 100%;
                                position: absolute;
                                top: 0;
                                left: 0;
                                background: $white;
                                border-radius: 50%;
                            }

                            a {
                                height: 100%;
                                display: flex;
                                align-items: center;
                                justify-content: center;
                                position: relative;
                            }
                        }
                    }
                `}
            </style>
        </>
    );
};
