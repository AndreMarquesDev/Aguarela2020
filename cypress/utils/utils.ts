import { Locale } from '../../types/Locale';
import {
    menuIconButtonDataTestId,
    nukaCarouselNextButtonDataTestId,
} from '../../utils/dataTestIds';
import { TextsInterface, textsPt, textsEn } from '../../utils/texts';

export const getLocalizedTexts = (locale: Locale): TextsInterface => {
    if (locale === Locale.Pt) {
        return textsPt;
    }

    return textsEn;
};

export const openMenuMobile = (isMobile: boolean, isFirefox: boolean): void => {
    if (isMobile) {
        cy.getByDataTestId(menuIconButtonDataTestId).click({ force: isFirefox });
    }
};

export const clickNextArrowButtonIfMobile = (
    isMobile: boolean,
    container: string,
    containerTitle: string
): void => {
    const isFirefox = Cypress.isBrowser('firefox');

    if (isMobile) {
        cy.getByDataTestIdParent(container, nukaCarouselNextButtonDataTestId).click({
            force: isFirefox,
        });

        cy.getByText(container, containerTitle).scrollIntoView();
    }
};
