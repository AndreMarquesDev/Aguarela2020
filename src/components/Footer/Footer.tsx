import type { Locale } from '../../types/Locale';
import type { Page } from '../../utils/pages';
import Link from 'next/link';
import React, { useContext } from 'react';
import { FaEnvelope, FaFacebookF, FaInstagram } from 'react-icons/fa';
import { tabletMediaQuery } from '../../styles/mediaQueries';
import { footerDataTestId } from '../../utils/dataTestIds';
import {
    aguarelaDigitalFacebookUrl,
    aguarelaDigitalInstagramUrl,
    andreMarquesDevWebsiteUrl,
} from '../../utils/urls';
import { TextsContext } from '../context/TextsContext';

export type FooterProps = {
    language: Locale;
};

export const Footer = ({ language }: FooterProps): JSX.Element => {
    const { texts } = useContext(TextsContext);

    return (
        <>
            <footer data-testid={footerDataTestId}>
                <div className="wrapper">
                    <p className="fontXS">
                        {texts.footerInfo}
                        {' '}
                        <a
                            className="animatedLink animatedLinkWhite"
                            href={andreMarquesDevWebsiteUrl}
                            rel="noreferrer"
                            target="_blank"
                        >
                            Andr√© Marques
                        </a>
                    </p>

                    <ul>
                        <li>
                            <a
                                aria-label="Aguarela instagram"
                                className="iconLink"
                                href={aguarelaDigitalInstagramUrl}
                                rel="noreferrer"
                                target="_blank"
                            >
                                <FaInstagram />
                            </a>
                        </li>
                        <li>
                            <a
                                aria-label="Aguarela facebook"
                                className="iconLink"
                                href={aguarelaDigitalFacebookUrl}
                                rel="noreferrer"
                                target="_blank"
                            >
                                <FaFacebookF />
                            </a>
                        </li>
                        <li>
                            <Link
                                aria-label="Go to contact page"
                                className="iconLink"
                                href={`/${language}/${'contact' as Page}`}
                                prefetch={false}
                            >
                                <FaEnvelope />
                            </Link>
                        </li>
                    </ul>
                </div>
            </footer>

            <style jsx>
                {`
                    footer {
                        background: var(--blue);
                        padding-top: 12.5rem;
                        padding-bottom: 12.5rem;
                    }

                    .wrapper {
                        display: flex;
                        justify-content: space-between;
                        align-items: center;

                        @media (${tabletMediaQuery}) {
                            flex-direction: column-reverse;
                        }
                    }

                    p {
                        color: var(--white);
                        text-transform: uppercase;

                        @media (${tabletMediaQuery}) {
                            margin-top: 25rem;
                            text-align: center;
                        }
                    }

                    ul {
                        display: flex;
                        justify-content: space-between;
                        align-items: center;

                        li {
                            width: 35rem;
                            height: 35rem;
                            position: relative;
                            text-align: center;
                            margin: 0 calc(25rem / 2);
                            color: var(--blue);

                            &:before {
                                content: '';
                                width: 100%;
                                height: 100%;
                                position: absolute;
                                top: 0;
                                left: 0;
                                background: var(--white);
                                border-radius: 50%;
                            }
                        }
                    }

                    :global(.iconLink) {
                        height: 100%;
                        display: flex;
                        align-items: center;
                        justify-content: center;
                        position: relative;
                    }
                `}
            </style>
        </>
    );
};
