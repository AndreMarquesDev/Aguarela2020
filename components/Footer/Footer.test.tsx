import '@testing-library/jest-dom';
import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import { Footer, FooterProps } from './Footer';
import { textsEn, textsPt } from '../../utils/texts';
import { MockTextsContext } from '../../utils/jest/MockTextsContext';

const baseProps: FooterProps = {
    language: 'pt',
};

const renderComponent = (newProps?: Partial<FooterProps>, isEnglish = false): RenderResult => {
    return render(
        <MockTextsContext isEnglish={isEnglish}>
            <Footer {...baseProps} {...newProps} />
        </MockTextsContext>
    );
};

describe('<Footer />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(textsPt.footerInfo)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = renderComponent({ language: 'en' }, true);

        expect(screen.getByText(textsEn.footerInfo)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
