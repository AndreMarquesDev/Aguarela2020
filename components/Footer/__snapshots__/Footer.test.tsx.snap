// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders properly 1`] = `
<div>
  <footer
    class="jsx-1577276803"
    data-testid="footer_section"
  >
    <div
      class="jsx-1577276803 wrapper"
    >
      <p
        class="jsx-1577276803"
      >
        Aguarela Digital - Developed by
         
        <a
          class="jsx-1577276803 animatedLink animatedLinkWhite"
          href="https://www.andremarquesdev.com"
          rel="noreferrer"
          target="_blank"
        >
          André Marques
        </a>
      </p>
      <ul
        class="jsx-1577276803"
      >
        <li
          class="jsx-1577276803"
        >
          <a
            aria-label="Aguarela instagram"
            class="jsx-1577276803"
            href="https://www.instagram.com/aguareladigital/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
        </li>
        <li
          class="jsx-1577276803"
        >
          <a
            aria-label="Aguarela facebook"
            class="jsx-1577276803"
            href="https://www.facebook.com/aguareladigitalpt"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              />
            </svg>
          </a>
        </li>
        <li
          class="jsx-1577276803"
        >
          <a
            aria-label="Go to contact page"
            class="jsx-1577276803"
            href="/pt/contact"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;

exports[`<Footer /> renders properly in English 1`] = `
<div>
  <footer
    class="jsx-1577276803"
    data-testid="footer_section"
  >
    <div
      class="jsx-1577276803 wrapper"
    >
      <p
        class="jsx-1577276803"
      >
        Aguarela Digital - Developed by
         
        <a
          class="jsx-1577276803 animatedLink animatedLinkWhite"
          href="https://www.andremarquesdev.com"
          rel="noreferrer"
          target="_blank"
        >
          André Marques
        </a>
      </p>
      <ul
        class="jsx-1577276803"
      >
        <li
          class="jsx-1577276803"
        >
          <a
            aria-label="Aguarela instagram"
            class="jsx-1577276803"
            href="https://www.instagram.com/aguareladigital/"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224.1 141c-63.6 0-114.9 51.3-114.9 114.9s51.3 114.9 114.9 114.9S339 319.5 339 255.9 287.7 141 224.1 141zm0 189.6c-41.1 0-74.7-33.5-74.7-74.7s33.5-74.7 74.7-74.7 74.7 33.5 74.7 74.7-33.6 74.7-74.7 74.7zm146.4-194.3c0 14.9-12 26.8-26.8 26.8-14.9 0-26.8-12-26.8-26.8s12-26.8 26.8-26.8 26.8 12 26.8 26.8zm76.1 27.2c-1.7-35.9-9.9-67.7-36.2-93.9-26.2-26.2-58-34.4-93.9-36.2-37-2.1-147.9-2.1-184.9 0-35.8 1.7-67.6 9.9-93.9 36.1s-34.4 58-36.2 93.9c-2.1 37-2.1 147.9 0 184.9 1.7 35.9 9.9 67.7 36.2 93.9s58 34.4 93.9 36.2c37 2.1 147.9 2.1 184.9 0 35.9-1.7 67.7-9.9 93.9-36.2 26.2-26.2 34.4-58 36.2-93.9 2.1-37 2.1-147.8 0-184.8zM398.8 388c-7.8 19.6-22.9 34.7-42.6 42.6-29.5 11.7-99.5 9-132.1 9s-102.7 2.6-132.1-9c-19.6-7.8-34.7-22.9-42.6-42.6-11.7-29.5-9-99.5-9-132.1s-2.6-102.7 9-132.1c7.8-19.6 22.9-34.7 42.6-42.6 29.5-11.7 99.5-9 132.1-9s102.7-2.6 132.1 9c19.6 7.8 34.7 22.9 42.6 42.6 11.7 29.5 9 99.5 9 132.1s2.7 102.7-9 132.1z"
              />
            </svg>
          </a>
        </li>
        <li
          class="jsx-1577276803"
        >
          <a
            aria-label="Aguarela facebook"
            class="jsx-1577276803"
            href="https://www.facebook.com/aguareladigitalpt"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
              />
            </svg>
          </a>
        </li>
        <li
          class="jsx-1577276803"
        >
          <a
            aria-label="Go to contact page"
            class="jsx-1577276803"
            href="/en/contact"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </footer>
</div>
`;
